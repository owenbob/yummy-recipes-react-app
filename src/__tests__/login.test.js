import React from 'react';
import ReactDOM from 'react-dom';
import { render,  configure,shallow } from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import expect from "expect";
import Adapter from 'enzyme-adapter-react-16';
import Login from '../components/login';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';


 configure({ adapter: new Adapter() });


describe('Login', () => {
    it('matches  snapshot', () => {
            const header = shallow(<Login />);
           expect(shallowToJson(header)).toMatchSnapshot();
        });

    it('renders without crashing', () => {
                const div = document.createElement('div');
                ReactDOM.render(
                <MuiThemeProvider>
                <Login />
                </MuiThemeProvider>, div);
        });
    
})