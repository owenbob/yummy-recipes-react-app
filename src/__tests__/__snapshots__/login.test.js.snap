// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches  snapshot 1`] = `
<div
  className="Login"
>
  <div
    className="center"
  >
    <Card
      expandable={false}
      expanded={null}
      initiallyExpanded={false}
    >
      <div
        className="inner"
      >
        <h1
          className="text"
        >
          Welcome To Yummy Recipes
        </h1>
        <form
          onSubmit={[Function]}
        >
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Username"
            fullWidth={false}
            hintText=" Enter your Username"
            multiLine={false}
            name="username"
            onChange={[Function]}
            rows={1}
            type="text"
            underlineShow={true}
            value=""
          />
          <br />
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Password"
            fullWidth={false}
            hintText=" Enter your Password"
            multiLine={false}
            name="password"
            onChange={[Function]}
            rows={1}
            type="password"
            underlineShow={true}
            value=""
          />
          <br />
          <RaisedButton
            disabled={false}
            fullWidth={false}
            label="Login"
            labelPosition="after"
            primary={true}
            secondary={false}
            style={Object {}}
            type="submit"
          />
          <h2
            className="text"
          >
            Not yet Registed?
          </h2>
          <RaisedButton
            disabled={false}
            fullWidth={false}
            href="/"
            label="Register"
            labelPosition="after"
            primary={true}
            secondary={false}
            style={Object {}}
          />
        </form>
      </div>
    </Card>
  </div>
</div>
`;
